server/routes/bookAVisit.ts

```diff
@@ -123,9 +125,50 @@ export default function routes(
         return closedVisit || visitor.restrictions.some(restriction => restriction.restrictionType === 'CLOSED')
       }, false)
       visitSessionData.visitRestriction = closedVisitVisitors ? 'CLOSED' : 'OPEN'
-
       visitSessionData.closedVisitReason = closedVisitVisitors ? 'visitor' : undefined
 
+      const closedVisitPrisoner = visitSessionData.prisoner.restrictions.reduce((closedVisit, restriction) => {
+        return closedVisit || restriction.restrictionType === 'CLOSED'
+      }, false)
+
+      if (!closedVisitVisitors && closedVisitPrisoner) {
```

Comment:
Could combine these into a ternary but that's up to you :) Means it's a single return statement.